lib/msf/core/post/process.rb

```diff
@@ -36,8 +36,10 @@ def has_pid?(pid)
   # Gets the `pid` and `name` of the processes on the remote system
   #
   def get_processes
-    if session_has_process_ext
+    if session.type == 'meterpreter'
       meterpreter_get_processes
+    elsif session.type == 'powershell'
+      powershell_get_processes
```

Comment:
The Powershell version using `Get-Process` is not including the `.exe` suffix of the name. This is a bit concerning because the Meterpreter and shell ones do. This would cause things like `pidof` to break which is expecting there to be an exact match. I couldn't find a way that would include the exe suffix in powershell that wasn't also throwing intermittent errors.

How about removing the Powershell one entirely, and using `shell_get_processes` for it? That one checks the platform internally and works for Windows as it is.

Reply:
This should work, I tried locally and tested with windows payload and for linux used powershell itself without getting any session because currently we don't have powershell payloads for non-windows platforms. :+1:

Reply:
```
>> get_processes
=> 
[{"name"=>"[System Process]", "pid"=>0},
 {"name"=>"System", "pid"=>4},
 {"name"=>"Registry", "pid"=>92},
...
...
...
 {"name"=>"svchost.exe", "pid"=>268},
 {"name"=>"svchost.exe", "pid"=>7828},
 {"name"=>"tasklist.exe", "pid"=>1152},
 {"name"=>"WmiPrvSE.exe", "pid"=>4536}]
```